ehsm_kms_service/enroll_app/enroll_msg.cpp

```diff
@@ -0,0 +1,1266 @@
+ * process will be send to eHSM-KMS msg4 json
+ *  p_msg4 : json
+ *      {
+ *          "mac": array(int),
+ *          "platform_info_blob": {
+ *              "nonce": {
+ *                  "rand": array(int)
+ *              },
+ *              "quote_verification_result" : int,
+ *              "qve_report_info": {
+ *                  "app_enclave_target_info": {
+ *                      "attributes": {
+ *                          "flags": uint64(string),
+ *                          "xfrm": uint64(string)
+ *                      },
+ *                      "config_id": array(int),
+ *                      "config_svn": int,
+ *                      "misc_select": int,
+ *                      "mr_enclave": {
+ *                          "m": array(int),
+ *                      },
+ *                      "reserved1": array(int),
+ *                      "reserved2": array(int),
+ *                      "reserved3": array(int)
+ *                  },
+ *                  "nonce": {
+ *                      "rand": array(int)
+ *                  },
+ *                  "qe_report": {
+ *                      "body": {
+ *                          "attributes": {
+ *                              "flags": uint64(string),
+ *                              "xfrm": uint64(string)
+ *                          },
+ *                          "config_id": array(int),
+ *                          "config_svn": int,
+ *                          "cpu_svn": {
+ *                              "svn": array(int)
+ *                          },
+ *                          "isv_ext_prod_id": array(int),
+ *                          "isv_family_id": array(int),
+ *                          "isv_prod_id": int,
+ *                          "isv_svn": int,
+ *                          "misc_select": int,
+ *                          "mr_enclave": {
+ *                              "m": array(int)
+ *                          },
+ *                          "mr_signer": {
+ *                              "m": array(int)
+ *                          },
+ *                          "report_data": {
+ *                              "d": array(int)
+ *                          },
+ *                          "reserved1": array(int),
+ *                          "reserved2": array(int),
+ *                          "reserved3": array(int),
+ *                          "reserved4": array(int)
+ *                      },
+ *                      "key_id": {
+ *                          "id": array(int)
+ *                      },
+ *                      "mac": array(int)
+ *                  }
+ *              }
+ *          },
+ *          "quote_type": int,
+ *          "secret": {
+ *              "payload": array(int),
+ *              "payload_size": int,
+ *              "payload_tag": array(int),
+ *              "reserved": array(int)
+ *          }
+ *      }
+ */
+enroll_status_t ra_proc_msg4(sample_ra_att_result_msg_t *tp_msg4, std::string *p_msg4)
+{
```

Comment:
marshal_msg4_to_json

Reply:
add put the marshal operations into a split file(marshal.cpp)

Reply:
It will be change